src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -0,0 +1,149 @@
+    {
+        var mapActionOperations = context.SyntaxProvider.CreateSyntaxProvider(
+            predicate: (node, _) => node is InvocationExpressionSyntax
+            {
+                Expression: MemberAccessExpressionSyntax
+                {
+                    Name: IdentifierNameSyntax
+                    {
+                        Identifier: { ValueText: var method }
+                    }
+                },
+                ArgumentList: { Arguments: { Count: 2 } args }
+            } && _knownMethods.Contains(method),
+            transform: (context, token) => context.SemanticModel.GetOperation(context.Node, token) as IInvocationOperation);
+
+        var endpoints = mapActionOperations
+            .Select((operation, _) => StaticRouteHandlerModelParser.GetEndpointFromOperation(operation))
+            .Where(endpoint => endpoint.Response.ResponseType == "string") // Only emit for signatures we support
+            .WithTrackingName("EndpointModel");
+
+        var genericThunks = endpoints.Select((endpoint, token) =>
+        {
+            var code = RequestDelegateGeneratorSources.GetGenericThunks(string.Empty);
+            return code;
+        });
+
+        var thunks = endpoints.Select((endpoint, _) => $@"            [{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)}] = (
+           (del, builder) =>
+            {{
+builder.Metadata.Add(new SourceKey{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)});
+            }},
+           (del, builder) =>
+            {{
+                var handler = ({StaticRouteHandlerModelEmitter.EmitHandlerDelegateType(endpoint)})del;
+                EndpointFilterDelegate? filteredInvocation = null;
+
+                if (builder.FilterFactories.Count > 0)
+                {{
+                    filteredInvocation = BuildFilterDelegate(ic =>
+                    {{
+                        if (ic.HttpContext.Response.StatusCode == 400)
+                        {{
+                            return System.Threading.Tasks.ValueTask.FromResult<object?>(Results.Empty);
+                        }}
+                        {StaticRouteHandlerModelEmitter.EmitFilteredInvocation()}
+                    }},
+                    builder,
+                    handler.Method);
+                }}
+
+                {StaticRouteHandlerModelEmitter.EmitRequestHandler()}
+                {StaticRouteHandlerModelEmitter.EmitFilteredRequestHandler()}
+
+                return filteredInvocation is null ? RequestHandler : RequestHandlerFiltered;
+            }}),
+");
+
+        var stronglyTypedEndpointDefinitions = endpoints.Select((endpoint, _) =>
+        {
+            var code = new StringBuilder();
+            code.AppendLine($"internal static Microsoft.AspNetCore.Builder.RouteHandlerBuilder {endpoint.HttpMethod}");
+            code.Append("(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints, string pattern, ");
```

Comment:
```suggestion
            code.Append(@"(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints, [StringSyntax(""Route"")] string pattern, ");
```

Reply:
Might need to update using statements

https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.stringsyntaxattribute?view=net-7.0